### PR TITLE
Restrict request quote emails

### PR DESCRIPTION
**Pull request by sderickson**
* If logged in => repopulate and lock email (and name if available)
* If logged out => disallow using an email for an existing account and display a validation message + login link

### CODE REVIEW

File: app/views/RequestQuoteView.coffee [hunk 1]
```diff
@@ -70,15 +87,27 @@ module.exports = class RequestQuoteView extends RootView
...
     @trialRequest.on 'sync', @onTrialRequestSubmit, @
     @trialRequest.on 'error', @onTrialRequestError, @
     me.setRole attrs.role.toLowerCase(), true
 
-  onTrialRequestError: ->
+  onTrialRequestError: (model, jqxhr) ->
+    if jqxhr.status is 409
+      @$('#email-form-group')
+        .addClass('has-error')
+        .append($('<div class="help-block error-help-block">User exists with this email. <a id="email-exists-login-link">Log in.</a>'))
```
Comment (differentmatt) [inline app/views/RequestQuoteView.coffee]: Localize `Log in.`?

File: app/views/RequestQuoteView.coffee [hunk 2]
```diff
@@ -45,6 +48,20 @@ module.exports = class RequestQuoteView extends RootView
...
+    if obj
+      @$('#other-education-level-checkbox').attr('checked', obj.otherChecked)
+      @$('#other-education-level-input').val(obj.otherInput)
+      forms.objectToForm(@$('form'), obj)
+
+  onChangeForm: ->
+    obj = forms.formToObject(@$('form'))
+    obj.otherChecked = @$('#other-education-level-checkbox').is(':checked')
+    obj.otherInput = @$('#other-education-level-input').val()
+    storage.save('request-quote-form', obj, 10)
```
Comment (differentmatt) [inline app/views/RequestQuoteView.coffee]: Why are we only saving these two items in storage on form change?

**Comment by differentmatt** [comment]
Done reviewing the code; couple minor things.  Did not run your changes, but can if necessary.

Ship it!

File: app/views/RequestQuoteView.coffee (hunk 2 above)
Comment (sderickson) [inline app/views/RequestQuoteView.coffee]: We're not, we're saving the whole thing, forms.formToObject gets everything else. The 'other' field has special logic (if checked, add string in input to array of strings for educationLevel property) so it doesn't get pulled automatically, either when backing up or on form submit, so in both cases it's [handled manually](https://github.com/codecombat/codecombat/blob/master/app/views/RequestQuoteView.coffee#L52-L54).

This is a good example of how abnormal forms lead to opaque logic. I'll add comments to make it clearer. A good general solution doesn't come to mind.

**Pull request merged**